client/landing/stepper/declarative-flow/flows/new-hosted-site-flow/new-hosted-site-flow.ts

```diff
@@ -171,46 +173,52 @@ const hosting: FlowV2 = {
 					return navigate( STEPS.PROCESSING.slug );
 
 				case STEPS.PROCESSING.slug: {
-					const siteId = providedDependencies.siteId || getSignupCompleteSiteID();
-					const siteSlug = providedDependencies.siteSlug || getSignupCompleteSlug();
-					const destinationParams: Record< string, string > = {
-						siteId,
-					};
-					if ( studioSiteId ) {
-						destinationParams[ 'redirect_to' ] = addQueryArgs( `/home/${ siteId }`, {
-							studioSiteId,
-						} );
-					} else if ( isWooPartner ) {
-						// For partners, we'll redirect to the WooCommerce admin page
-						destinationParams[
-							'redirect_to'
-						] = `https://${ siteSlug }/wp-admin/admin.php?page=wc-admin`;
-					}
-					// Purchasing Business or Commerce plans will trigger an atomic transfer, so go to stepper flow where we wait for it to complete.
-					const destination = addQueryArgs( '/setup/transferring-hosted-site', destinationParams );
-
-					// If the product is a free trial, record the trial start event for ad tracking.
-					if ( planCartItem && isFreeHostingTrial( planCartItem?.product_slug ) ) {
-						recordFreeHostingTrialStarted( flowName );
-					}
-
-					if ( providedDependencies.goToCheckout ) {
-						persistSignupDestination( destination );
-						setSignupCompleteSlug( providedDependencies?.siteSlug );
-						setSignupCompleteFlowName( flowName );
-
-						couponCode && resetCouponCode();
-						return window.location.assign(
-							addQueryArgs(
-								`/checkout/${ encodeURIComponent(
-									( providedDependencies?.siteSlug as string ) ?? ''
-								) }`,
-								{ redirect_to: destination, coupon: couponCode }
-							)
+					if ( providedDependencies.processingResult === ProcessingResult.SUCCESS ) {
```

Comment:
I think we can early exit to the error step here instead of wrapping the whole success logic within a conditional block.

Reply:
The problem with that is that the processing step submits SUCCESS, NO_ACTION, and FAILURE. So we'd have to handle the NO_ACTION as well. And I'm not sure we should go to error on NO_ACTION.

Reply:
Right. Let's keep what we have right now.